TLM/TLM/UI/SubTools/LaneConnectorTool.cs

```diff
@@ -948,6 +965,7 @@ public LaneConnectorTool(TrafficManagerTool mainTool)
             hoveredLaneEnd = null;
             stayInLaneMode = StayInLaneMode.None;
             RefreshCurrentNodeMarkers();
+            MainTool.RequestOnscreenDisplayUpdate();
```

Comment:
if you have to call MainTool.RequestOnscreenDisplayUpdate() in so many places its a sign of badly organized code and also increases the likelyhood of human error. would it help if you do one of the following: 
- use `try{ ... }finally {MainTool.RequestOnscreenDisplayUpdate() }` 
- override `OnToolUpdate()` and do it in every `OnToolUpdate()` call.
-  do it in TrafficManagerTool.OnToolUpdate() in response to click event 
- do it in OnToolGUI in response to any  mouse/key up/down event.

Reply:
There aren't clear transition changes in legacy subtools, so i have to hunt each one down.
When the tool is upgraded, the update calls will only be located in state change handlers.
Yes the legacy code is badly organized and is very prone to errors that's why it is renamed to legacy and marked obsolete.
No i am not doing this outside in OnToolUpdate or OnToolGUI, that happens every frame.

Reply:
there is no harm done if you call `RequestOnscreenDisplayUpdate()` too many times is there?

Reply:
I mean in response to any user event so that it does not reduce FPS.

Reply:
It will reduce FPS. Every time you call update it clears the hint panel and fills new labels for help, and also resizes the main menu and all labels in it.

Reply:
I think full refactoring can come later, maybe as a v11.6.0 task? There's already lots of changes in 11.5.0 for users to test.